core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java

```diff
@@ -1642,7 +1648,13 @@ public long saveEmPcrDilution(emPCRDilution dilution) throws IOException {
   @Override
   public long saveLibrary(Library library) throws IOException {
     if (libraryStore != null) {
-      return libraryStore.save(library);
+      SampleAdditionalInfo info = library.getSample().getSampleAdditionalInfo();
+      if (LibraryPropagationRule.validate(library,
+          libraryPropagationRuleDao.getLibraryPropagationRulesByClass(info == null ? null : info.getSampleClass()))) {
+        return libraryStore.save(library);
+      } else {
+        throw new IOException("Invalid propagation.");
```

Comment:
When using the single sample/library ui to generate a library from a detailed sample a stack trace is displayed when the propagation validation fails.